src/main/java/seedu/partyplanet/logic/parser/DeleteCommandParser.java

```diff
@@ -62,10 +63,15 @@ private DeleteContactWithTagCommand createDeleteContactWithTagCommand(List<Strin
     private DeleteContactCommand createDeleteContactCommand(String args) throws ParseException {
         String[] strIndexes = args.split("\\s+");
         List<Index> indexes = new ArrayList<>();
+        Set<Integer> intIndices = new HashSet<>();
 
         for (String s : strIndexes) {
             Index index = ParserUtil.parseIndex(s);
+            if (intIndices.contains(index.getZeroBased())) {
+                continue;
+            }
             indexes.add(index);
+            intIndices.add(index.getZeroBased());
         }
 
         return new DeleteContactCommand(indexes);
```

Comment:
For marking event as done (in the event branch)
I simply used `list.contains()` since that uses `.equals()` to check.
May be slightly less efficient time complexity wise, but could be more readable. Which do yall prefer?
If prefer this then LGTM, i can update the other

```suggestion
    private DeleteContactCommand createDeleteContactCommand(String args) throws ParseException {
        String[] strIndexes = args.split("\\s+");
        List<Index> indexes = new ArrayList<>();

        for (String s : strIndexes) {
            Index index = ParserUtil.parseIndex(s);
            if (!indexes.contains(index)) {
                indexes.add(index);
            }
        }

        return new DeleteContactCommand(indexes);
```

Reply:
This should be better:
1. Assuming in typical use scenarios, the number of unique indices specified is not large (pretty unlikely)
2. Since the application does not require strong performance optimization, readability should be a priority

Will apply suggestion and merge.